R/dsl.R

```diff
@@ -192,104 +307,169 @@ dropdown <- function(name,
+#' @param type Type of the menu. Look at https://semantic-ui.com/collections/menu.html for all possiblities.
+#' @param disable_ui_class If TRUE menu is created without ui class - used when UI menu is child of other ui class.
+#'
+#' @examples
+#' library(shiny)
+#' library(shiny.semantic)
+#'
+#' ui <- function() {
+#'   shinyUI(
+#'     semanticPage(
+#'       title = "My page",
+#'       suppressDependencies("bootstrap"),
+#'       uimenu(menu_item("Menu"),
+#'              uidropdown(
+#'                "Action",
+#'                dropdown_menu(
+#'                  menu_header(uiicon("file"), "File", is_item = FALSE),
+#'                  menu_item(uiicon("wrench"), "Open"),
+#'                  menu_item(uiicon("upload"), "Upload"),
+#'                 menu_item(uiicon("remove"), "Upload"),
+#'                  menu_divider(),
+#'                  menu_header(uiicon("user"), "User", is_item = FALSE),
+#'                  menu_item(uiicon("add user"), "Add"),
+#'                  menu_item(uiicon("remove user"), "Remove")),
+#'                type = "",
+#'                name = "unique_name",
+#'                is_menu_item = TRUE),
+#'              menu_item(uiicon("user"), "Profile", is_link = TRUE, link = "#index", item_feature = "active"),
+#'              menu_item("Projects", is_link = TRUE, link = "#projects"),
+#'              menu_item(uiicon("users"), "Team"),
+#'              dropdown_menu(menu_item(uiicon("add icon"), "New tab"), type = "right"))
+#'     )
+#'   )
+#' }
+#'
+#' server <- shinyServer(function(input, output) {
+#' })
+#'
+#' shinyApp(ui = ui(), server = server)
 #'
 #' @export
-tabset <- function(tabs,
-                   id = generate_random_id("menu"),
-                   menu_class = "top attached tabular",
-                   tab_content_class = "bottom attached segment") {
-  identifiers <- replicate(length(tabs),
-                           list(id = generate_random_id("tab")),
-                           simplify = FALSE)
-  id_tabs <- purrr::map2(identifiers, tabs, ~ c(.x, .y))
+uimenu <- function(..., type = "", disable_ui_class = FALSE) {
+  class <- "menu"
+  if (!disable_ui_class) {
+    class <- paste("ui", class)
+  }
+  div(class = paste(class, type),
+      list(...))
+}
+
+#' Create Semantic UI Menu Item
+#'
+#' This creates a menu item using Semantic UI
+#'
+#' @param ... Content of the menu item: text, icons or labels to be displayed.
+#' @param item_feature If required, add additional item feature like 'active', 'header', etc.
+#' @param style Style of the item, e.g. "text-align: center".
+#' @param is_link If TRUE menu item is created with 'a' tag - ortherwise with 'div' tag (default).
+#' @param link If is_link is TRUE, determines to what the menu item should be linked. Ignored when is_link is FALSE.
+#'
+#' @export
+menu_item <- function(..., item_feature = "", style = NULL, is_link = FALSE, link = NULL) {
```

Comment:
Please remove is_link from args. Please change link to href. When href != NULL => use tags$a.